examples/gcoap/client.c

```diff
@@ -223,24 +214,50 @@ int gcoap_cli_cmd(int argc, char **argv)
         printf("CoAP open requests: %u\n", open_reqs);
         printf("Configured Proxy: ");
         if (_proxied) {
+#ifdef SOCK_HAS_IPV6
             char addrstr[IPV6_ADDR_MAX_STR_LEN];
-            printf("[%s]:%u\n",
-                   ipv6_addr_to_str(addrstr,
-                                    (ipv6_addr_t *) &_proxy_remote.addr.ipv6,
-                                    sizeof(addrstr)),
-                   _proxy_remote.port);
+#else
+            char addrstr[IPV4_ADDR_MAX_STR_LEN];
+#endif
+
+            switch (_proxy_remote.family) {
+#ifdef SOCK_HAS_IPV4
+            case AF_INET:
+                inet_ntop(AF_INET, _proxy_remote.addr.ipv4, addrstr, sizeof(addrstr));
```

Comment:
Why the `switch`?
```suggestion
                inet_ntop(_proxy_remote.family, _proxy_remote.addr.ipv4, addrstr, sizeof(addrstr));
```

Reply:
The switch is `#ifdef`-friendly. I know that condition-based compilation is not strictly necessary (at the moment) for IPv4 because it will be present in `_sock_tl_ep` for IPv4 and IPv6 socks, but can we be absolutely sure that this will be always the case? Looks like an implementation detail for me that we should not rely on.

Reply:
Oh and the `printf` format string is different for IPv6, so we would at least need a `if-else` on `_proxy_remote.family` anyway

Reply:
I don't understand - the `inet_ntop()` is entirely address-agnostic.

Reply:
> Oh and the `printf` format string is different for IPv6, so we would at least need a `if-else` on `_proxy_remote.family` anyway

You will still end up with less code in the end.

Reply:
Maybe I'm just super confused but `inet_ntop` will call `ipv4_addr_to_str` for example which expects a byte buffer of size 4 (`ipv4_addr_t`). We can't pass `_proxy_remote` as you propose because it is a `sock_udp_ep_t` which is a typedef'd `_sock_tl_ep` where we need to access either `addr.ipv4` or `addr.ipv6`.

Reply:
Won't`&_proxy_remote.addr` work?

Reply:
Ah I didn't see you made a pointer out of the union, `&_proxy_remote.addr` will work. Feels a bit hacky to me though, but it has the advantage that we don't care about existing and non-existing members in the union.

Reply:
Changed as suggested